src/actions/secrets.js

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { getCredentials, deleteCredential, createCredential } from '../api';
+import { getSelectedNamespace } from '../reducers';
+
+export function fetchSecretsSuccess(data, namespace) {
+  return {
+    type: 'SECRETS_FETCH_SUCCESS',
+    data,
+    namespace
+  };
+}
+
+// to be used when 'all namespaces' option is ready
+// export function fetchSecretsSuccess(data) {
+//   return {
+//     type: 'SECRETS_FETCH_SUCCESS',
+//     data
+//   };
+// }
+
+export function fetchSecrets({ namespace } = {}) {
+  return async (dispatch, getState) => {
+    dispatch({ type: 'SECRETS_FETCH_REQUEST' });
+    let secrets;
+    try {
+      const selectedNamespace = namespace || getSelectedNamespace(getState());
+      secrets = await getCredentials(selectedNamespace);
+      dispatch(fetchSecretsSuccess(secrets, selectedNamespace));
+      // dispatch(fetchSecretsSuccess(secrets));
+    } catch (e) {
+      const error = new Error('Could not fetch secrets');
+      dispatch({ type: 'SECRETS_FETCH_FAILURE', error });
+    }
+    return secrets;
+  };
+}
+
+export function deleteSecret(id) {
+  return async (dispatch, getState) => {
+    dispatch({ type: 'SECRET_DELETE_REQUEST' });
+    try {
+      const namespace = getSelectedNamespace(getState());
+      await deleteCredential(id, namespace);
+      dispatch({ type: 'SECRET_DELETE_SUCCESS', id });
+      dispatch(fetchSecrets());
+    } catch (e) {
+      const error = new Error(`Could not delete secret ${id}`);
+      dispatch({ type: 'SECRET_DELETE_FAILURE', error });
+    }
+  };
+}
+
+/* istanbul ignore next */
+export function createSecret(postData, namespace) {
+  return async dispatch => {
+    dispatch({ type: 'SECRET_CREATE_REQUEST' });
+    try {
+      await createCredential(postData, namespace);
+      dispatch(fetchSecrets());
+    } catch (e) {
+      const error = new Error(
+        `Could not create secret ${postData.name} in namespace ${namespace}`
+      );
+      dispatch({ type: 'SECRET_CREATE_FAILURE', error });
+    }
+  };
+}
+
+export function clearNotification() {
+  return { type: 'CLEAR_NOTIFICATION' };
```

Comment:
How is this used? If it's used to dismiss an InlineNotification or similar is there a reason this can't be handled in component state or using the built-in close button of the notification component?

Is it specific to secrets? If so, the type should reflect this.

Reply:
So the use case is basically to show the any error with an inline notification in the dashboard for a specific amount of time & once time has passed call this action which basically all is does is fall in the errorMessage reducer which ultimately just adds null to the error variable & that is what makes it disappear in the dashboard.

Reply:
`ToastNotification` has an optional `timeout` prop if that's what you're trying to achieve, however it might make sense to allow the user to dismiss the notification manually as we do with some of the others. Either way we shouldn't really need this action.

Reply:
how can i go about nullifying the error value in the store??

Reply:
So far the pattern has been that the next request / success would reset it. If we really do need this one to timeout and it's controlled by a value in the store then we'll need to keep the action but it should be renamed to make it clear which error / resource type it's associated with.